### PR TITLE
Reduce allfont to be a metapackage

### PR DESCRIPTION
**Pull request by tiziodcaio**
Better management and cleaner environment
I don't know if i cheated with the yaml... Is it valid?

### CODE REVIEW

**Comment by mirkobrombin** [comment]
Thank you for your contribution. It looks fine, we need to check every single font manifest to make sure the install process is aligned with what it is now in allfonts (which is the most up to date).

**Comment by mirkobrombin** [comment]
Well done! Now those edits need to be tested.

**Comment by tiziodcaio** [comment]
Can I ask how to test everything?

**Comment by tiziodcaio** [comment]
Hmm there is a problem with the meta: it stops on times32 without considering the others on the list... there is a limit to 9 dependencies maybe? It seems strange...

```
➤ LOCAL_DEPENDENCIES=/home/daniele/git/dependencies/ bottles

(bottles:25539): Handy-WARNING **: 13:58:45.530: Using GtkSettings:gtk-application-prefer-dark-theme together with HdyStyleManager is unsupported. Please use HdyStyleManager:color-scheme instead.
(13:58:45) INFO Using local dependencies repository at /home/daniele/git/dependencies/ 
(13:58:45) INFO Bottles Started! 
(13:58:46) INFO Performing Bottles checks... 
(13:58:46) INFO Dxvks found:
         - dxvk-1.10.1
 
(13:58:46) INFO Vkd3ds found:
         - vkd3d-proton-2.6
 
(13:58:46) INFO Nvapis found:
         - dxvk-nvapi-v0.5.4
 
(13:58:46) INFO Latencyflexs found:
         - latencyflex-v0.1.0
 
(13:58:46) INFO Runners found:
         - sys-wine-7.6
         - caffe-7.6
         - vaniglia-21.2.0-cx
 
(13:58:46) INFO Bottles found:
         - test
         - Gaming
 
(13:58:51) WARNING Cannot find states.yml file for bottle: [test] 

(bottles:25539): Handy-CRITICAL **: 13:58:51.708: hdy_stackable_box_set_visible_child_name: assertion 'contains_child' failed
(13:58:55) INFO Installing dependency [allfonts] in bottle [test]. 
(13:58:55) INFO Installing dependency [arial32] in bottle [test]. 
(13:58:55) WARNING File [arial32.exe] already exists in temp, skipping. 
(13:58:55) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Arial] and Data: [Arial.TTF] in test registry 
(13:58:55) INFO Using WINE Registry CLI -- add 
wineserver: using server-side synchronization.
002c:err:wineboot:process_run_key Error running cmd L"C:\\windows\\system32\\winemenubuilder.exe -r" (2).
(13:58:56) INFO reg: The operation completed successfully

 
(13:58:56) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Arial Bold] and Data: [Arialbd.TTF] in test registry 
(13:58:57) INFO Using WINE debug tool -- get_processes 
(13:58:57) INFO Using WINE Registry CLI -- add 
(13:58:57) INFO reg: The operation completed successfully

 
(13:58:57) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Arial Italic] and Data: [Ariali.TTF] in test registry 
(13:58:57) INFO Using WINE debug tool -- get_processes 
(13:58:58) INFO Using WINE Registry CLI -- add 
(13:58:58) INFO reg: The operation completed successfully

 
(13:58:58) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Arial Bold Italic] and Data: [Arialbi.TTF] in test registry 
(13:58:58) INFO Using WINE debug tool -- get_processes 
(13:58:58) INFO Using WINE Registry CLI -- add 
(13:58:58) INFO reg: The operation completed successfully

 
(13:58:58) INFO Setting Key: [Installed_Dependencies] to [['arial32']] for bottle: [test]… 
(13:58:58) INFO Setting Key: [arial32] to [False] for bottle: [test]… 
(13:58:58) INFO Installing dependency [arialb32] in bottle [test]. 
(13:58:58) WARNING File [arialb32.exe] already exists in temp, skipping. 
(13:58:58) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Arial Black] and Data: [AriBlk.TTF] in test registry 
(13:58:59) INFO Using WINE debug tool -- get_processes 
(13:58:59) INFO Using WINE Registry CLI -- add 
(13:58:59) INFO reg: The operation completed successfully

 
(13:58:59) INFO Setting Key: [Installed_Dependencies] to [['arial32', 'arialb32']] for bottle: [test]… 
(13:58:59) INFO Setting Key: [arialb32] to [False] for bottle: [test]… 
(13:58:59) INFO Installing dependency [comic32] in bottle [test]. 
(13:58:59) WARNING File [comic32.exe] already exists in temp, skipping. 
(13:58:59) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Comic Sans MS] and Data: [Comic.TTF] in test registry 
(13:59:00) INFO Using WINE debug tool -- get_processes 
(13:59:00) INFO Using WINE Registry CLI -- add 
(13:59:00) INFO reg: The operation completed successfully

 
(13:59:00) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Comic Sans MS Bold] and Data: [Comicbd.TTF] in test registry 
(13:59:01) INFO Using WINE debug tool -- get_processes 
(13:59:01) INFO Using WINE Registry CLI -- add 
(13:59:01) INFO reg: The operation completed successfully

 
(13:59:01) INFO Setting Key: [Installed_Dependencies] to [['arial32', 'arialb32', 'comic32']] for bottle: [test]… 
(13:59:01) INFO Setting Key: [comic32] to [False] for bottle: [test]… 
(13:59:01) INFO Installing dependency [courie32] in bottle [test]. 
(13:59:01) WARNING File [courie32.exe] already exists in temp, skipping. 
(13:59:01) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Courier New] and Data: [cour.ttf] in test registry 
(13:59:01) INFO Using WINE debug tool -- get_processes 
(13:59:02) INFO Using WINE Registry CLI -- add 
(13:59:02) INFO reg: The operation completed successfully

 
(13:59:02) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Courier New Bold Italic] and Data: [courbi.ttf] in test registry 
(13:59:02) INFO Using WINE debug tool -- get_processes 
(13:59:02) INFO Using WINE Registry CLI -- add 
(13:59:03) INFO reg: The operation completed successfully

 
(13:59:03) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Courier New Italic] and Data: [couri.ttf] in test registry 
(13:59:03) INFO Using WINE debug tool -- get_processes 
(13:59:03) INFO Using WINE Registry CLI -- add 
(13:59:03) INFO reg: The operation completed successfully

 
(13:59:03) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Georgia Bold Italic] and Data: [Georgiaz.TTF] in test registry 
(13:59:04) INFO Using WINE debug tool -- get_processes 
(13:59:04) INFO Using WINE Registry CLI -- add 
(13:59:04) INFO reg: The operation completed successfully

 
(13:59:04) INFO Setting Key: [Installed_Dependencies] to [['arial32', 'arialb32', 'comic32', 'courie32']] for bottle: [test]… 
(13:59:04) INFO Setting Key: [courie32] to [False] for bottle: [test]… 
(13:59:04) INFO Installing dependency [tahoma32] in bottle [test]. 
(13:59:04) WARNING File [IELPKTH.CAB] already exists in temp, skipping. 
/home/daniele/.local/share/bottles/temp/IELPKTH.CAB: WARNING; possible 5592 extra bytes at end of file.
/home/daniele/.local/share/bottles/temp/tahoma32/tahomabd.ttf: Permission denied
/home/daniele/.local/share/bottles/temp/IELPKTH.CAB: WARNING; possible 5592 extra bytes at end of file.
/home/daniele/.local/share/bottles/temp/tahoma32/tahoma.ttf: Permission denied
Copying tahoma.ttf to /home/daniele/.local/share/bottles/temp//tahoma.ttf
Copying tahomabd.ttf to /home/daniele/.local/share/bottles/temp//tahomabd.ttf
(13:59:04) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Tahoma] and Data: [tahoma.ttf] in test registry 
(13:59:05) INFO Using WINE debug tool -- get_processes 
(13:59:05) INFO Using WINE Registry CLI -- add 
(13:59:05) INFO reg: The operation completed successfully

 
(13:59:05) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Tahoma Bold] and Data: [tahomabd.ttf] in test registry 
(13:59:06) INFO Using WINE debug tool -- get_processes 
(13:59:06) INFO Using WINE Registry CLI -- add 
(13:59:06) INFO reg: The operation completed successfully

 
(13:59:06) INFO Setting Key: [Installed_Dependencies] to [['arial32', 'arialb32', 'comic32', 'courie32', 'tahoma32']] for bottle: [test]… 
(13:59:06) INFO Setting Key: [tahoma32] to [False] for bottle: [test]… 
(13:59:06) INFO Installing dependency [times32] in bottle [test]. 
(13:59:06) WARNING File [times32.exe] already exists in temp, skipping. 
(13:59:06) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Times New Roman] and Data: [Times.TTF] in test registry 
(13:59:07) INFO Using WINE debug tool -- get_processes 
(13:59:07) INFO Using WINE Registry CLI -- add 
(13:59:07) INFO reg: The operation completed successfully

 
(13:59:07) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Times New Roman Bold] and Data: [Timesbd.TTF] in test registry 
(13:59:07) INFO Using WINE debug tool -- get_processes 
(13:59:07) INFO Using WINE Registry CLI -- add 
(13:59:08) INFO reg: The operation completed successfully

 
(13:59:08) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Times New Roman Italic] and Data: [Timesi.TTF] in test registry 
(13:59:08) INFO Using WINE debug tool -- get_processes 
(13:59:08) INFO Using WINE Registry CLI -- add 
(13:59:08) INFO reg: The operation completed successfully

 
(13:59:08) INFO Adding Key: [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Fonts] with Value: [Times New Roman Bold Italic] and Data: [Timesbi.TTF] in test registry 
(13:59:09) INFO Using WINE debug tool -- get_processes 
(13:59:09) INFO Using WINE Registry CLI -- add 
(13:59:09) INFO reg: The operation completed successfully

 
(13:59:09) INFO Setting Key: [Installed_Dependencies] to [['arial32', 'arialb32', 'comic32', 'courie32', 'tahoma32', 'times32']] for bottle: [test]… 
(13:59:09) INFO Setting Key: [times32] to [False] for bottle: [test]… 
(13:59:09) INFO Setting Key: [Installed_Dependencies] to [['arial32', 'arialb32', 'comic32', 'courie32', 'tahoma32', 'times32', 'allfonts']] for bottle: [test]… 
(13:59:09) INFO Setting Key: [allfonts] to [True] for bottle: [test]…
```

**Comment by mirkobrombin** [comment]
There are no limits but it may stop processing if one fails

**Comment by tiziodcaio** [comment]
immagine
immagine
Now it works 😃

**Comment by tiziodcaio** [comment]
The last detail.... Tahoma misteriously was extracted with wrong permissions (read only for owners)... So I had to copy the files again in temp to fix permissions, otherwise I got errors

**Comment by mirkobrombin** [comment]
It didn’t give me this issue 🤔 can you open an issue to track the problem?

**Comment by tiziodcaio** [comment]
They are missing georgi32 and impact32 🤔

**Comment by tiziodcaio** [comment]
But it's allfonts, they should be added... otherwise they would not be all 😄

File: Fonts/allfonts.yml [hunk 1]
```diff
@@ -3,276 +3,17 @@ Description: All Microsoft and Adobe essential fonts
...
-  dest: temp/comic32
-
-- action: cab_extract
-  file_name: courie32.exe
-  url: https://mirrors.kernel.org/gentoo/distfiles/courie32.exe
-  file_checksum: 4e412c772294403ab62fb2d247d85c60
-  file_size: 646368
-  dest: temp/courie32
-
-- action: cab_extract
```
[the remark this answers is not in the document]
Comment (mirkobrombin) [reply Fonts/allfonts.yml:44]: that's a big question 🤔 georgi and impact should not be so useful.. ya ok, add them too

**Comment by tiziodcaio** [comment]
#81

**Pull request merged**